### ISSUE TITLE
502 Response error while creating a new github webhook

### ISSUE THREAD

**Issue by akhil97** [sdaschner/tekton-argocd-example#3]
![image](https://user-images.githubusercontent.com/19953362/166181999-1abbc545-0bb6-4b14-9575-1927b6c44e95.png)
I was trying to create a new github webhook using the link :- https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads . I tried out the steps mentioned in the README but I am getting an 502 response status code back

**Comment by sdaschner** [comment]
Like shown in the video, you need to make sure that the endpoint works in your cluster. You can first try that yourself via the command line (with `curl`, for instance), and then once you know that everything works add it to GitHub - that should be easier for debugging

**Comment by akhil97** [comment]
Can you tell me specifically which endpoints do I need to look at?

**Comment by sdaschner** [comment]
Well, depends what you added in your cluster, for example [such a gateway](https://github.com/sdaschner/tekton-argocd-example/blob/main/production/gateway.yaml).

Btw, did you watch my [video course](https://blog.sebastian-daschner.com/entries/cloud-native-ci-cd-tekton-argocd-video-course) on this topic? It should be explained there :)

**Comment by akhil97** [comment]
Yes, I have gone through your video. I am facing trouble trying to expose the event listener outside the cluster for receiving trigger requests from GitHub web hook. I have used ngrok's http tunnel for sending the trigger request. I wanted to know if you have used the same approach for creating webhooks

**Comment by sdaschner** [comment]
Ah ok, no I have used the Istio gateways, like shown in the video and the code. In general, you can use any way you want to expose the triggers, but they need to be accessible from the internet (you can test that upfront), otherwise it won't work with GH
